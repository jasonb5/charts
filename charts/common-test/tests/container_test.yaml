suite: test container
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should overwrite defaults
    set:
      workload: deployment

      args:
        - infinity
      command:
        - /bin/sleep
      env:
        CONFIG: /config/config.txt
        PORT: "{{ .Values.service.ports.default.port }}"
        RETRIES: 8
      envFrom:
        default:
          type: configmap
        credentials:
          type: secret
      image:
        pullPolicy: Always
      lifecycle:
        postStart:
          exec:
            command: ["curl", "..."]
      livenessProbe:
        tcpSocket:
          port: "{{ .Values.service.ports.default.port }}"
      readinessProbe:
        tcpSocket:
          port: "{{ .Values.service.ports.default.port }}"
      resources:
        limits:
          cpu: 2
      restartPolicy: Always
      securityContext:
        runAsUser: 65536
        runAsGroup: 65536
      startupProbe:
        tcpSocket:
          port: "{{ .Values.service.ports.default.port }}"
      workingDir: /app

      service:
        ports:
          default:
            port: 80
    asserts:
      - matchSnapshot:
          path: spec.template.spec.containers[0]
        documentIndex: 0
      - hasDocuments:
          count: 2
  - it: should pass
    set:
      workload: deployment
    asserts:
      - matchSnapshot:
          path: spec.template.spec.containers[0]
        documentIndex: 0
      - hasDocuments:
          count: 1
