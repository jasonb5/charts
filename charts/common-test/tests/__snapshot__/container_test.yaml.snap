should overwrite defaults:
  1: |
    args:
      - infinity
    command:
      - /bin/sleep
    env:
      - name: CONFIG
        value: /config/config.txt
      - name: PORT
        value: 80
      - name: RETRIES
        value: 8
    envFrom:
      - secretRef:
          name: RELEASE-NAME-common-test-credentials
      - configMapRef:
          name: RELEASE-NAME-common-test
    image: busybox:0.1.0
    imagePullPolicy: Always
    lifecycle:
      postStart:
        exec:
          command:
            - curl
            - '...'
    livenessProbe:
      tcpSocket:
        port: "80"
    name: common-test
    ports:
      - containerPort: 80
        name: common-test
        protocol: TCP
    readinessProbe:
      tcpSocket:
        port: "80"
    resources:
      limits:
        cpu: 2
    restartPolicy: Always
    securityContext:
      runAsGroup: 65536
      runAsUser: 65536
    startupProbe:
      tcpSocket:
        port: "80"
    workingDir: /app
should pass:
  1: |
    image: busybox:0.1.0
    name: common-test
